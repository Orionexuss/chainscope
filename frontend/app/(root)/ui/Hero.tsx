import Link from "next/link";

const Hero = () => {
  return (
    <div className="flex h-screen flex-col gap-7 px-3 py-[8vh] text-center text-white sm:gap-[4vh] lg:py-[1vw]">
      <div>
        <h1 className="font-(family-name:--koulen) text-[13vw] leading-[1.1] sm:text-[10vw] sm:leading-none lg:text-8xl xl:text-9xl gradient-text">
          FOLLOW THE MOST <br /> PROFITABLE WALLETS
        </h1>
      </div>
      <div>
        <h2 className="font-(family-name:--koulen) text-[6vw] sm:text-3xl xl:text-4xl">
          SEE WHAT TOP WALLETS ARE DOING BEFORE THE CROWD
        </h2>
      </div>
      <div>
        <p className="max-w-140 font-(family-name:--abel) mx-auto text-xl text-[#DADADA] sm:text-xl lg:text-2xl">
          Spot trends early, analyze smart money behavior, and make informed
          decisions backed by on-chain data.
        </p>
      </div>
      <div className="relative mx-auto mt-6 w-fit md:mt-3">
        <button className="relative mb-2 me-2 inline-flex items-center justify-center rounded-lg bg-gradient-to-br from-green-400 to-blue-600 p-0.5 font-medium hover:text-white group">
          <Link
            href={`/auth/register`}
            className="font-(family-name:--koulen) relative cursor-pointer rounded-md bg-transparent px-9 py-3.5 text-3xl transition-all duration-75 ease-in sm:bg-cyan-950 sm:group-hover:bg-transparent group-hover:dark:bg-transparent"
          >
            <span>TRY IT &#62;</span>
          </Link>
        </button>
      </div>
    </div>
  );
};

export default Hero;
