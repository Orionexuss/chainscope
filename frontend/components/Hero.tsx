const Hero = () => {
  return (
    <div className="flex flex-col justify-center py-8 text-center text-white">
      <div className="font-(family-name:--koulen)">
        <h1 className="text-8xl my-9">
          FOLLOW THE MOST <br /> PROFITABLE WALLETS
        </h1>
        <h2 className="text-3xl">SEE WHAT TOP WALLETS ARE DOING BEFORE THE CROWD</h2>
      </div>
      <div>
        <p className="text-[#DADADA] text-2xl font-(family-name:--abel) mt-7">Spot trends early, analyze smart money behavior, and <br /> make informed decisions backed by on-chain data.</p>
      </div>
    </div>
  );
};

export default Hero;
