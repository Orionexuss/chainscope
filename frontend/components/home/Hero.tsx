const Hero = () => {
  return (
    <div className="hero-container">
      <div>
        <h1 className="hero-heading gradient-text">
          FOLLOW THE MOST <br /> PROFITABLE WALLETS
        </h1>
      </div>
      <div>
        <h2 className="hero-subheading">
          SEE WHAT TOP WALLETS ARE DOING BEFORE THE CROWD
        </h2>
      </div>
      <div>
        <p className="hero-paragraph">
          Spot trends early, analyze smart money behavior, and make informed
          decisions backed by on-chain data.
        </p>
      </div>
      <div className="relative w-fit mx-auto mt-6 md:mt-3">
        <button className="hero-button group">
          <span className="hero-button-text">TRY IT &#62;</span>
        </button>
      </div>
    </div>
  );
};

export default Hero;
