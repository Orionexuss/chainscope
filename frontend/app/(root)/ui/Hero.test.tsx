import { render, screen } from "@testing-library/react";
import Hero from "./Hero";
import "@testing-library/jest-dom";

describe("Hero Component", () => {
  beforeEach(() => {
    render(<Hero />);
  });

  it("renders the main heading", () => {
    const mainHeading = screen.getByRole("heading", { level: 1 });
    expect(mainHeading).toHaveTextContent(
      /FOLLOW THE MOST PROFITABLE WALLETS/i,
    );
  });

  it("renders the subheading", () => {
    const subHeading = screen.getByText(
      /SEE WHAT TOP WALLETS ARE DOING BEFORE THE CROWD/i,
    );
    expect(subHeading).toBeInTheDocument();
  });

  it("renders the paragraph text", () => {
    const paragraph = screen.getByText(
      /Spot trends early, analyze smart money behavior, and make informed/i,
    );
    expect(paragraph).toBeInTheDocument();
  });

  it("renders the button with correct text", () => {
    const button = screen.getByText(/TRY IT >/i);
    expect(button).toBeInTheDocument();
  });

  it("renders the link with correct href", () => {
    const link = screen.getByRole("link", { name: /TRY IT >/i });
    expect(link).toHaveAttribute("href", "/auth/register");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<Hero />);
    expect(asFragment()).toMatchSnapshot();
  });
});
