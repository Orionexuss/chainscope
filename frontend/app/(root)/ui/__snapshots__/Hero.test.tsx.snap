// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex h-screen flex-col gap-7 px-3 py-[8vh] text-center text-white sm:gap-[4vh] lg:py-[1vw]"
  >
    <div>
      <h1
        class="font-(family-name:--koulen) text-[13vw] leading-[1.1] sm:text-[10vw] sm:leading-none lg:text-8xl xl:text-9xl gradient-text"
      >
        FOLLOW THE MOST 
        <br />
         PROFITABLE WALLETS
      </h1>
    </div>
    <div>
      <h2
        class="font-(family-name:--koulen) text-[6vw] sm:text-3xl xl:text-4xl"
      >
        SEE WHAT TOP WALLETS ARE DOING BEFORE THE CROWD
      </h2>
    </div>
    <div>
      <p
        class="max-w-140 font-(family-name:--abel) mx-auto text-xl text-[#DADADA] sm:text-xl lg:text-2xl"
      >
        Spot trends early, analyze smart money behavior, and make informed decisions backed by on-chain data.
      </p>
    </div>
    <div
      class="relative mx-auto mt-6 w-fit md:mt-3"
    >
      <button
        class="relative mb-2 me-2 inline-flex items-center justify-center rounded-lg bg-gradient-to-br from-green-400 to-blue-600 p-0.5 font-medium hover:text-white group"
      >
        <a
          class="font-(family-name:--koulen) relative cursor-pointer rounded-md bg-transparent px-9 py-3.5 text-3xl transition-all duration-75 ease-in sm:bg-cyan-950 sm:group-hover:bg-transparent group-hover:dark:bg-transparent"
          href="/auth/register"
        >
          <span>
            TRY IT &gt;
          </span>
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
